### PR TITLE
Overview: Visually show a minimum amount of storage used

### PR DESCRIPTION
**Pull request by chad1008**
Related to 8460-gh-Automattic/dotcom-forge

## Proposed Changes

- Updates the Storage usage meter on `/overview` to always show a minimum amount of storage used, rather than rendering an empty bar when used storage is very low
- The original issue called for 1% as a starting figure, but after some further conversation we settled on 2.5%, which is where it's set currently. I've kept the following screenshots included below for reference, and I'll update this issue if we change the value again before merging.
- Removed the red color declaration on the progress bar. `ProgressBar` itself will likely be getting a revamp, and this ensures we pick those colors up when it happens. The default blue also fits the overall aesthetic of the page better anyway.

Some visual samples:

| Header | Header |
|--------|--------|
| trunk | image |
| 1% | image |
| 2% | image |
| 3% | image | 

## Why are these changes being made?

An empty bar can be confusing for a user, because it's just one solid color. Is it full? Empty? It it a progress bar? By always displaying a small amount of 'used' storage, it's more visually clear that the user has a long way to go before they reach their limit.

## Testing Instructions

- Load `/overview` for any site with little to no storage used
- Notice the empty storage progress bar
- Apply this patch
- Notice the new minimum being displayed
- Switch to a site with more storage being used, or experiment by hardcoding a higher value [in place of](https://github.com/Automattic/wp-calypso/blob/2ca6bcca43689b55fae05697d7f30b05364d15d4/client/hosting-overview/components/plan-storage-bar.tsx#L19) `mediaStorage.storageUsedBytes`
- Confirm that when the amount of storage used exceeds the current minimum display value, the actual usage takes over an fills the progress bar appropriately

### CODE REVIEW

**Comment by nuriapenya** [comment]
Looks cool! I'd suggest avoiding the flat indicator and, instead, using the full circle once we have relevant data to show.

Here's an idea of how that could go:

screen 2024-07-26 at 15 32 34

Figma link: R8yjT7mjG0Uw8h2GPyIY6R-fi-3912_5590

**Comment by chad1008** [comment]
This PR is close to complete, pending finalization of the design aspects. Looking at the related P2 thread, it sounds like we'll be going with a blue progress bar, potentially shifting to a darker red at 81%+

@nuriapenya, I saw your note about lightening the background color of the bar itself to improve contrast with that red. Is that something you think would be best to do just in these implementations of `ProgressBar` on `/overview`?  Or something we want to apply to the component itself so it's consistent across Calypso? I'm happy to wrangle it either way, based on your design wisdom 🙂

**Comment by chad1008** [comment]
I've added `DO NOT MERGE` to this for now... just to remind me that we're still waiting on finalizing colors and such.

**Comment by nuriapenya** [comment]
@chad1008, correct! We will be updating the progress bar's logic a bit and choosing different values for the current red. I would love to roll this across the product 🙌🏽  I know it might increase a bit the scope, but we would then ensure consistency across. 

If you want, I can open an issue dedicated to these design tweaks :) so you can keep this one focused only on the logic of the minimum amount shown.

**Comment by chad1008** [comment]
Updated to reflect feedback in another conversation (pdKhl6-45r-p2#comment-7097) where a minimum of 2.5 was recommended. It looks a little too close to the wider 3% for my tastes, but that's just a personal preference:
image

Worth noting: the visual rendering is dependent on screen size, as that impacts the width of the progress bar as a whole. The above screenshot is with a full-screen browser window at 1728px wide. Below, we have the same 2.5% minimum at some other breakpoints:

| 1285px | 1280px |
|--------|--------|
| image | image |
| _Looks smaller, closer to a 1% minimum_ | _Similar to full width, perhaps a pixel or two closer to the 3% look_ |

And for the mobile impact, here's emulating a Google Pixel 7 at 412px wide:
image


My takeaway here reinforces my instinct not to over analyze the minimum, unless we want to get into logic that tries to keep the visual look of a mimimum amount rendering consistent across screen sizes... which is doable, but feels a little like scope-creep for a nice-to-have in my mind. Happy to dig into it though if it's desirable!

**Comment by chad1008** [comment]
Adding a note for our future selves, I know we've discussed having a default blue progress bar that changes to red at a set point when storage is running low (80% is what I remember being tossed around).

While that functionality isn't built into `ProgressBar` itself (yet?) there is an implementation of it [already tucked into the `PlanStorageBar` component](https://github.com/Automattic/wp-calypso/blob/c7042ecda75b3368cba8c973ef7976ef6b24eae2/client/blocks/plan-storage/bar.jsx#L38-L42). It has a "warn" and "alert" threshold that add CSS classes we can style off of very easily, we'd just need to change the color values. "Alert" is, conveniently, already set to 80%.

So, to sum up... we can build on what this progress bar implementation already has, or if this gets baked directly into `ProgressBar` itself, we can remove some of the logic from this specific case to simplify the code here.

**Comment by nuriapenya** [comment]
Fantastic work 🙌🏽 

>  It has a "warn" and "alert" threshold that add CSS classes we can style off of very easily, we'd just need to change the color values. "Alert" is, conveniently, already set to 80%.

It sounds like we could update the component right away. The threshold is already set (exactly, we also agreed at 80%); we just need to turn the current default to our Blueberry #3858E9 and turn the "Alert" switch to the current red.

**Comment by chad1008** [comment]
> It sounds like we could update the component right away. The threshold is already set (exactly, we also agreed at 80%); we just need to turn the current default to our Blueberry #3858E9 and turn the "Alert" switch to the current red.

Perfect. I'll make those updates in this PR. If we end up implementing similar logic in `ProgressBar` directly in the future, we can also move this implementation over in a followup 👍

**Comment by chad1008** [comment]
Slight wrinkle, it turns out we actually have two _different_ `PlanStorageBar` components, and we aren't using the one with the built-in alert level in this case. I missed this initially because I mistook the file name `bar.tsx` as a subcomponent and didn't catch the matching name on the component itself.

Not a huge deal, adding an 80% alert to the hosting overview's `PlanStorageBar` is a very easy addition. It's duplication of code, which I don't love, but the two components are styled differently enough that it makes more sense than trying two switch. I am, however, going to add an issue to Letero's backlog to investigate these two components, where they're used, and how we might combine them into one component with props to accommodate different use cases.

In the meantime, Hosting Overview now renders in Blueberry, up until it breaks the 80% barrier, at which point it switches to the red that the bar used previously. That should cover design concerns, so we can move on to code review!

**Comment by nuriapenya** [comment]
> add an issue to Letero's backlog to investigate these two components, where they're used, and how we might combine them into one component with props to accommodate different use cases.

Cool 👍🏽 I was about to ask why we have two different components, but let's wait for the investigation then; no rush to solve this right now. Thanks, Chad!

File: client/hosting/overview/components/plan-storage-bar.tsx [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
 
-	const percent = Math.min(
-		Math.round( ( ( mediaStorage.storageUsedBytes / mediaStorage.maxStorageBytes ) * 1000 ) / 10 ),
-		100
+	// Ensure that the displayed usage is never fully empty to avoid a confusing UI
+	const displayedUsage = Math.max(
+		MINIMUM_DISPLAYED_USAGE,
+		Math.round( ( ( storageUsedBytes / maxStorageBytes ) * 1000 ) / 10 )
 	);
+	const percent = Math.min( displayedUsage, 100 );
```
Comment (nightnei) [inline client/hosting/overview/components/plan-storage-bar.tsx]: It's not critical, but I propose rewriting it a bit, since at a glance it looks not clear and difficult to read a lot of deep math operations.
What is your opinion about the next?

```suggestion
	let usagePercent = Math.round( ( ( storageUsedBytes / maxStorageBytes ) * 1000 ) / 10 );

	// Ensure that the displayed usage is never fully empty to avoid a confusing UI
	usagePercent = Math.max( MINIMUM_DISPLAYED_USAGE, usagePercent );

	// Ensure that somehow it's not bigger than 100
	usagePercent = Math.min( usagePercent, 100 );
```

**Review by nightnei — APPROVED** [review]
LGTM and works as expected 👍
Just left 1 comment regarding readability.

Screenshot 2024-08-06 at 18 14 19
Screenshot 2024-08-06 at 18 15 19
Screenshot 2024-08-06 at 18 16 02

File: client/hosting/overview/components/plan-storage-bar.tsx (hunk 1 above)
Comment (chad1008) [reply client/hosting/overview/components/plan-storage-bar.tsx]: I love it! I thought about breaking up that math too, but sometimes I think I'm over-simplifying and making things more verbose than they need to be. Thanks for helping me dial in on the sweet spot!

File: client/hosting/overview/components/plan-storage-bar.tsx [hunk 2]
```diff
@@ -39,7 +48,10 @@ const PlanStorageBar: FC< PropsWithChildren< Props > > = ( { children, mediaStor
 				</span>
 			</div>
 
-			<ProgressBar color="var(--studio-red-30)" value={ percent } total={ 100 } compact={ false } />
+			<div className={ classes }>
```
Comment (fredrikekelund) [inline client/hosting/overview/components/plan-storage-bar.tsx:51]:
```suggestion
			<div className={ clsx( 'plan-storage__bar', {
				'is-alert': usagePercent > ALERT_PERCENT,
			} ) }>
```
Nit, but for future reference, let's default to inlining `clsx` calls directly in the `className` attribute. That's more conventional.

File: client/hosting/overview/components/style.scss [hunk 3]
```diff
@@ -208,6 +209,15 @@ $card-padding: 24px;
 	font-size: $font-body-extra-small;
 	line-height: 16px;
 	margin-top: 30px;
+
+	.plan-storage__bar .progress-bar__progress {
+		background-color: $blueberry-color;
```
Comment (fredrikekelund) [inline client/hosting/overview/components/style.scss:214]:
```suggestion
		background-color: var(--theme-highlight-color);
```
This seems to achieve the same effect, and using a global variable for a theme-level color is preferable. 

@nightnei, I think I saw some PR where you had also defined a local `$blueberry-color` variable.

@chad1008, could you spin up a PR to replace those instances with `var(--theme-highlight-color)` (unless you see a good reason not to)?
